django/db/models/sql/query.py

```diff
@@ -1418,7 +1460,7 @@ def setup_joins(self, names, opts, alias, can_reuse=None, allow_many=True):
             reuse = can_reuse if join.m2m else None
             alias = self.join(connection, reuse=reuse)
             joins.append(alias)
-        return final_field, targets, opts, joins, path
+        return final_field, targets, opts, joins, path, final_transformer
```

Comment:
This is getting out of hand, especially considering callers ignoring so many of the return values. Does it make sense to create a type to model these values? At a minimum I think we should consider namedtuple.

Reply:
This was done as #8840